dev-tools/mage/dockerbuilder.go

```diff
@@ -141,13 +145,47 @@ func (b *dockerBuilder) dockerSave(tag string) error {
 	// Save the container as artifact
 	outputFile := b.OutputFile
 	if outputFile == "" {
-		outputTar, err := b.Expand(defaultBinaryName + ".docker.tar")
+		outputTar, err := b.Expand(defaultBinaryName + ".docker.tar.gz")
 		if err != nil {
 			return err
 		}
 		outputFile = filepath.Join(distributionsDir, outputTar)
 	}
-	if err := sh.Run("docker", "save", "-o", outputFile, tag); err != nil {
+	var stderr bytes.Buffer
+	cmd := exec.Command("docker", "save", tag)
+	cmd.Stderr = &stderr
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	if err = cmd.Start(); err != nil {
+		return err
+	}
+
+	err = func() error {
+		f, err := os.Create(outputFile)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		w := gzip.NewWriter(f)
+		defer w.Close()
+
+		_, err = io.Copy(w, stdout)
+		if err != nil {
+			return err
+		}
+		return nil
+	}()
+	if err != nil {
+		return err
+	}
+
+	if err = cmd.Wait(); err != nil {
+		if errmsg := strings.TrimSpace(stderr.String()); errmsg != "" {
+			err = errors.Wrap(errors.New(errmsg), err.Error())
```

Comment:
I'd expect this to be the reverse `err = errors.Wrap(err, errmsg)`

Reply:
In general I'd say the same, but then it'd print `<command stderr>: <command error>`, what I find a bit counterintuitive for this case. I prefer it printing first the command error, and then the command stderr output, e.g:
```
Error: failed building heartbeat type=docker for platform=linux/amd64: failed to save docker as artifact: exit status 1: Error response from daemon: reference does not exist
```
Where `exit status 1` is the error message in `err` and `Error response from daemon: reference does not exist` the message in `errmsg` that comes from stderr.

Reply:
ah makes sense, thanks 👍